src/main/java/com/_up/megastore/services/implementations/ProductService.java

```diff
@@ -57,12 +57,20 @@ private String saveProductImage(MultipartFile multipartFile) {
     private Product getVariantOf(UUID variantOfId) {
         return variantOfId != null ? findProductByIdOrThrowException(variantOfId) : null;
     }
-    public ProductResponse updateProduct(UpdateProductRequest updateProductRequest, MultipartFile multipartFile){
-        Category category = categoryService.findCategoryByIdOrThrowException(UpdateProductRequest.categoryId());
-        Product variantOf = getVariantOf(updateProductRequest.variantOfId());
+    public ProductResponse updateProduct(UUID productId, UpdateProductRequest updateProductRequest, MultipartFile multipartFile){
+        Product product = this.findProductByIdOrThrowException(productId);
+        Category category = categoryService.findCategoryByIdOrThrowException(updateProductRequest.categoryId());
+        Product variantOf = this.getVariantOf(updateProductRequest.variantOfId());
```

Comment:
La idea está bien, pero hay un problema. En el UpdateProductRequest el 'variantOfId' no está validado por jakarta. Esto significa que el usuario puede incluirlo o no (si desea sobreescribir la variante o no).

Imaginate, entonces, que un usuario hace un update sin especificar el variantOfId (o escribiéndolo mal). Lo que va a pasar es que el producto, que antes apuntaba a otro producto, ahora va a pasar a ser _null_. Esto no debería ocurrir, sino que la variante debería ser exactamente la misma. Lo que sugeriría yo es agregar la siguiente validación:

```suggestion
    if (updateProductRequest.variantOfId() != null) {
        Product variantOf = this.getVariantOf(updateProductRequest.variantOfId());
        product.setVariantOf(variantOf);
    }
```
Que podría extraerse a su propio método quizás (ifVariantOfIdExistsUpdateProductVariantOf). Esto mismo podría aplicarse al Category si sacamos la validación @NotNull en la request.